pgmpy/models/BayesianModel.py

```diff
@@ -738,6 +747,72 @@ def predict(self, data):
                 pred_values[k].append(v)
         return pd.DataFrame(pred_values, index=data.index)
 
+    def predict_probability(self, data):
```

Comment:
@ankurankan The two methods use different methods of Inference (query and map_query). I can use maximize in query to calculate predict but that would be useless as we already have map_query for that. Therefore the common part would just consist of initializing Inference.